crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -569,6 +604,205 @@ pub fn if_with_same_arms(checker: &mut Checker, stmt: &Stmt, parent: Option<&Stm
     }
 }
 
+fn should_proceed_main(test: &Expr, body: &[Stmt], orelse: &[Stmt]) -> bool {
+    if let ExprKind::Compare {
+        left,
+        ops,
+        comparators,
+    } = &test.node
+    {
+        if let ExprKind::Name { .. } = &left.node {
+            if ops.len() == 1 && ops[0] == Cmpop::Eq {
+                if comparators.len() == 1 {
+                    if let ExprKind::Constant { .. } = &comparators[0].node {
+                        if body.len() == 1 {
+                            if let StmtKind::Return { .. } = &body[0].node {
+                                if orelse.len() == 1 {
+                                    if let StmtKind::If { .. } = &orelse[0].node {
+                                        return true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    false
```

Comment:
Nit: You could make use of Rust's pattern matching to remove some of the nesting but it's not necessarily more readable

```suggestion
if let ExprKind::Name { .. } = &left.node {
            if let [Cmpop::Eq] = ops.as_slice() {
                if let [Expr {
                    node: ExprKind::Constant { .. },
                    ..
                }] = comparators.as_slice()
                {
                    if let [Stmt {
                        node: StmtKind::Return { .. },
                        ..
                    }] = body
                    {
                        if let [Stmt {
                            node: StmtKind::If { .. },
                            ..
                        }] = orelse
                        {
                            return true;
                        }
                    }
                }
            }
        }
```

or even

```rust
    if let ExprKind::Compare {
        left,
        ops,
        comparators,
    } = &test.node
    {
        return matches!(
            (
                &left.node,
                ops.as_slice(),
                comparators.as_slice(),
                body,
                orelse
            ),
            (
                ExprKind::Name { .. },
                [Cmpop::Eq],
                [Expr {
                    node: ExprKind::Constant { .. },
                    ..
                }],
                [Stmt {
                    node: StmtKind::Return { .. },
                    ..
                }],
                [Stmt {
                    node: StmtKind::If { .. },
                    ..
                }]
            )
        );
    }
    false
```